reconcile/utils/terrascript_aws_client.py

```diff
@@ -1863,8 +1861,16 @@ def populate_tf_resource_s3(self, spec):
         }
         values["policy"] = json.dumps(policy, sort_keys=True)
         values["depends_on"] = self.get_dependencies([user_tf_resource])
-        tf_resource = aws_iam_user_policy(identifier, **values)
-        tf_resources.append(tf_resource)
+        tf_aws_iam_policy = aws_iam_policy(identifier, **values)
+        tf_resources.append(tf_aws_iam_policy)
+
+        tf_user_policy_attachment = aws_iam_user_policy_attachment(
```

Comment:
One thing to note here is that we are mixing resource invocation styles. Older code in this file follows this pattern:

```
values = {}
values['resource-property']
...
tf_aws_iam_policy = aws_iam_policy(identifier, **values)
```

Where as our style is more HCL-like:

```
tf_aws_iam_policy_attachment = aws_iam_user_policy_attachment(
                identifier,
                user=identifier,
                policy_arn=f"{{{tf_aws_iam_policy.arn}}}",
                depends_on=self.get_dependencies([user_tf_resource, tf_aws_iam_policy]),
            )
tf_resources.append(tf_aws_iam_policy_attachment)
```

Personally, I think the second style is much easier to read and makes objects look more like HCL, which is welcome in my opinion.

I don't have any direct actions out of this comment, just a note. If I had it my way, I would make everything look more like HCL and get rid of building value dictionaries line by line.

Really, if I had it my way, every provider would be a module written in HCL and instantiated in each provider creation method.